docs/development/CodingStyle.md

```diff
@@ -0,0 +1,253 @@
+(Adding a statement to a branch will brake the logic if the braces are forgotten and otherwise make the PR longer).
+
+##Spaces
+Use a space after (most) keywords.  The notable exceptions are sizeof, typeof, alignof, and __attribute__, which look somewhat like functions (and are usually used with parentheses).
+So use a space after these keywords:
+```
+if, switch, case, for, do, while
+```
+but not with sizeof, typeof, alignof, or __attribute__.  E.g.,
+```
+s = sizeof(struct file);
+```
+When declaring pointer data or a function that returns a pointer type, the preferred use of '*' is adjacent to the data name or function name and not adjacent to the type name.  Examples:
+```
+char *linux_banner;
+memparse(char *ptr, char **retptr);
+char *match_strdup(substring_t *s);
+```
+Use one space around (on each side of) most binary and ternary operators, such as any of these:
+```
+=  +  -  <  >  *  /  %  |  &  ^  <=  >=  ==  !=  ?  :
+```
+but no space after unary operators:
+```
+&  *  +  -  ~  !  sizeof  typeof  alignof  __attribute__  defined
+```
+no space before the postfix increment & decrement unary operators:
+```
+++  --
+```
+no space after the prefix increment & decrement unary operators:
+```
+++  --
+```
+and no space around the '.' and "->" structure member operators.
+
+'*' and '&', when used for pointer and reference, shall have no space between it and the following variable name.
+
+#typedef
+enums with a count should have that count declared as the last item in the enumeration list,
+so that it is automatically maintained, e.g.:
+```
+typedef enum {
+    PID_CONTROLLER_MW23 = 0,
+    PID_CONTROLLER_MWREWRITE,
+    PID_CONTROLLER_LUX_FLOAT,
+    PID_COUNT
+} pidControllerType_e;
+```
+It shall not be calculated afterwards, e.g. using PID_CONTROLLER_LUX_FLOAT + 1;
+
+typedef struct definitions should include the struct name, so that the type can be forward referenced, that is in:
+```
+typedef struct motorMixer_s {
+    float throttle;
+...
+    float yaw;
+} motorMixer_t;
+```
+the motorMixer_s name is required.
+
+#Variables
+
+##Naming
+Generally, descriptive lowerCamelCase names are preferred for function names, variables, arguments, etc.
+For configuration variables that are user accessible via CLI or similar, all_lowercase with underscore is preferred.
+
+Variable names should be nouns.
+
+Simple temporary variables with a very small scope may be short where it aligns with common practice.
+Such as "i" as a temporary counter in a "for" loop, like "for (uint_8 i = 0; i < 4; i++)".
+Using "temporaryCounter" in that case would not improve readability.
+
+##Declarations
+Avoid global variables.
+
+Variables should be declared at the top of the smallest scope where the variable is used.
+One blank line should follow the declaration(s).
+
+Variables with limited use may be declared at the point of first use. It makes PR-review easier (but that point is lost if the variable is used everywhere anyway).
+
```

Comment:
Been looking again on the switch case.
I guess you referring to situations where different case need a similar local variable and the compiler will catch a name clash.
If so, I'd say it's not anything for a style guide. The compiler will find it and anyone can figure out a work-round easily. To always use such extra braces would look ugly.